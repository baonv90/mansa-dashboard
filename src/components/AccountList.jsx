import React, { useCallback } from 'react'
import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  border-radius: .5rem;
  padding: .5rem;
  margin: 0 .5rem;
  background: ${({ isSelected }) => isSelected ?  'var(--primary-color)' : 'var(--grey-color)'};
  color: ${({ isSelected }) => isSelected ? 'white' : 'var(--text-color)'};
  min-width: 120px;
  min-height: 60px;
  cursor: pointer;
  align-items: center;
  justify-content: center;
`;

const AccountType = styled.span`
  font-size: .6rem;
  line-height: .6rem;
  font-weight: 400;
`;

const AccountNumber = styled.span`
  font-weight: 600;
`;


const AccountList = ({ isSelected, setSelectedTab, index, ...data }) => {
  const { account_number, account_type } = data;

  const onClick = useCallback(() => {
    setSelectedTab(index)
  }, [setSelectedTab, index]);

  return (
    <Container
      onClick={onClick}
      isSelected={isSelected}
    >
      <AccountType>{account_type}</AccountType>
      <AccountNumber>{account_number}</AccountNumber>
    </Container>
  )
}

export default AccountList