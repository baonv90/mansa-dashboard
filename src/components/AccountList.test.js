import renderer from 'react-test-renderer';
import AccountList, { Container } from './AccountList';

describe('component > AccountList', () => {
  it('should match snapshot', () =>  {
    const component = renderer.create(
      <AccountList />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot()
  });

  it('should call set selected tab onClick', () =>  {
    const props = {
      setSelectedTab: jest.fn()
    };
    const component = renderer.create(
      <AccountList {...props} />,
    );

    const el = component.root.findByType(Container);
    el.props.onClick();

    expect(props.setSelectedTab).toHaveBeenCalled();
  });
});