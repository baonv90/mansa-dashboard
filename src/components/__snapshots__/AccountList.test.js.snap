// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountList should match snapshot 1`] = `
<div
  className="sc-bdnxRM gkzNBE"
  onClick={[Function]}
>
  <span
    className="sc-gtsrHT dUFbuC"
  />
  <span
    className="sc-dlnjwi jHzDDr"
  />
</div>
`;
